MoreLinq.Test/SingleUseEnumerable.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MoreLinq.Test
+{
+    public sealed class SingleUseEnumerable<T> : IEnumerable<T>
```

Comment:
Come to think of it, this whole class is redundant with [`TestingSequence`](https://github.com/morelinq/MoreLINQ/blob/88163c3ec71536cad8961a76cb0c564692f4a1e5/MoreLinq.Test/TestingSequence.cs):

> Sequence that asserts whether its iterator has been disposed when it is disposed itself _and also whether `GetEnumerator()` is called exactly once or not._